hostpolicy/api/permissions.py

```diff
@@ -37,6 +37,9 @@ def has_permission(self, request, view):
 
         # Find out which labels are attached to this role
         role_labels = HostPolicyRole.objects.filter(name=view.kwargs['name']).values_list('labels__name',flat=True)
+        if not any(role_labels):
+            # if the role doesn't have any labels, there's no possibility of access at this point
+            return False
```

Comment:
Don't see how this should cause a security issue. Also a simple `if not role_labels` should suffice.

Reply:
TLDR: `any(label in [None] for label in [None])` is the issue

Reply:
Ah, makes sense and not at the same time. Returning `[None]` instead of an empty list was a bit unexpected.